File: clients/ios/SsdidDrive/SsdidDrive/Domain/Repository/WebAuthnRepository.swift
import Foundation

/// WebAuthn login begin result
struct WebAuthnBeginResult {
    let optionsJson: String
    let challengeId: String
}

/// Repository for WebAuthn/Passkey authentication operations
protocol WebAuthnRepository: AnyObject {
    /// Begin WebAuthn login (passkey assertion)
    func loginBegin(email: String?) async throws -> WebAuthnBeginResult

    /// Complete WebAuthn login with assertion
    func loginComplete(challengeId: String, assertionData: [String: Any]) async throws -> User

    /// Get user's credentials
    func getCredentials() async throws -> [UserCredential]

    /// Rename a credential
    func renameCredential(credentialId: String, name: String) async throws -> UserCredential

    /// Delete a credential
    func deleteCredential(credentialId: String) async throws
}

File: clients/android/app/src/main/kotlin/com/securesharing/domain/repository/WebAuthnRepository.kt
package com.securesharing.domain.repository

import com.securesharing.domain.model.User
import com.securesharing.domain.model.UserCredential
import com.securesharing.util.Result

interface WebAuthnRepository {
    suspend fun loginBegin(email: String?): Result<WebAuthnBeginResult>
    suspend fun loginComplete(
        challengeId: String,
        assertionJson: String,
        prfOutput: String?
    ): Result<User>
    suspend fun getCredentials(): Result<List<UserCredential>>
    suspend fun renameCredential(credentialId: String, name: String): Result<UserCredential>
    suspend fun deleteCredential(credentialId: String): Result<Unit>
}

data class WebAuthnBeginResult(
    val optionsJson: String,
    val challengeId: String
)
